src/core/qgslabelingengine.cpp

```diff
@@ -371,6 +373,10 @@ void QgsLabelingEngine::run( QgsRenderContext &context )
 
     context.expressionContext().setFeature( lf->feature() );
     context.expressionContext().setFields( lf->feature().fields() );
+    if ( lf->symbol() )
+    {
+      symbolScope = QgsExpressionContextUtils::updateSymbolScope( lf->symbol(), symbolScope );
+    }
     lf->provider()->drawLabel( context, label );
   }
```

Comment:
Don't forget to pop the symbolScope here. You'll likely also need this in the new draw label background loop too.

Reply:
@nyalldawson , do you want the scope to be appended and popped before / after loop or at every iteration?

Reply:
No, after the final iteration only